### PR TITLE
backport proxy and custom timeout support to console v4.11

### PR DESCRIPTION
**Pull request by kim-tsao**
Signed-off-by: Kim Tsao <ktsao@redhat.com>

### What does this PR do?:

Backports the changes to a dependency release supported by console 4.11.  Fix was verified in 4.12 by the console team

### Which issue(s) this PR fixes:

https://github.com/devfile/api/issues/926

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [ ] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation 

   

- [ ] Client Impact

  

- [ ] Gosec scans

### CODE REVIEW

File: pkg/devfile/parser/parse.go [hunk 1]
```diff
@@ -81,6 +81,8 @@ type ParserArgs struct {
 	Context context.Context
 	// K8sClient is the Kubernetes client instance used for interacting with a cluster
 	K8sClient client.Client
+	// HTTPTimeout overrides the request and response timeout values for reading a parent devfile reference from the registry.  If a negative value is specified, the default timeout will be used.
+	HTTPTimeout *int
```
Comment (yangcao77) [inline pkg/devfile/parser/parse.go:85]: any particular reason we make it a pointer?

Comment (kim-tsao) [reply pkg/devfile/parser/parse.go:85]: yes, it's to determine if this value is unset or not.  If unset, we will use the default value of 30s

File: README.md [hunk 2]
```diff
@@ -20,20 +20,28 @@ The function documentation can be accessed via [pkg.go.dev](https://pkg.go.dev/g
...
-		DefaultNamespace:  defaultNamespace,
-		Context:           context,
-		K8sClient:         client,
-	}
+        Path:              path,
+        FlattenedDevfile:  &flattenedDevfile,
+        RegistryURLs:      registryURLs,
+        DefaultNamespace:  defaultNamespace,
+        Context:           context,
+        K8sClient:         client,
```
[the remark this answers is not in the document]
Comment (kim-tsao) [reply README.md:28]: yes, good catch

[the remark this answers is not in the document]
Comment (kim-tsao) [reply README.md:28]: actually, that's just an example usage `HTTPTimeout` is optional so we can leave it out

**Review by yangcao77 — APPROVED** [review]

**Pull request merged**